### PR TITLE
Convert vala-linter as a reusable library

### PR DESCRIPTION
**Pull request by esodan**
This WIP, enable linter library to be installable and re-usable when installed and when used as subproject.

This will enable GVls to use it as a library to enable send diagnostics lint messages to the client, helping developers to match elementary code style.

The plan is that, once linter messages are integrated with GVls and demostrating it is useful, this branch could be merged, unless you consider this work could be useful for any one else.

### CODE REVIEW

**Comment by esodan** [comment]
Ok. This is now ready.

GVls has integrated linter in a wip/linter branch, waiting for this MR.

Please merge it.

**Comment by pantor** [comment]
Hi there! Maybe I'm a little bit out of the loop, but what is GVls?

File: data/vala-linter-1.0.pc.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+datadir=${prefix}/share
+includedir=${prefix}/include
+
+Name: @PROJECT_NAME@
+Description: GTK+ widget library for SVG objects using GSVG
```
Comment (pantor) [inline data/vala-linter-1.0.pc.in:8]: Please update description and URL.

**Comment by esodan** [comment]
GVls is the GNOME Vala Language Server, providing LSP support for IDE's having them and a client to add support for LSP to IED doesn't. https://gitlab.gnome.org/esodan/gvls

**Comment by esodan** [comment]
@pantor any progress here? I would like to use upstream instead of my fork in GVls and others IED that will integrate GVls to.

File: data/vala-linter-1.0.pc.in [hunk 2]
```diff
@@ -0,0 +1,13 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=@libdir@
+datadir=${prefix}/share
+includedir=${prefix}/include
+
+Name: vala-linter-1.0
+Description: Lqibrary prividing a Vala code files checker for code-style errors
```
Comment (pantor) [inline data/vala-linter-1.0.pc.in:8]: Some typos here.

**Comment by pantor** [comment]
I‘m afraid this might be a little bit over my head. Is it ok to set the minimum libvala version to 0.40?

Maybe @btkostner can help?

**Comment by btkostner** [comment]
Sorry I can't be a bigger help, but I'm not big into desktop development.

File: data/vala-linter-1.0.pc.in (hunk 2 above)
Comment (esodan) [reply data/vala-linter-1.0.pc.in:8]: First two lines there are not typos. but the line at Description, fixed.

Comment (esodan) [reply data/vala-linter-1.0.pc.in:8]: Version of libvala will be set to `>= 0.40.4` because is the version available in Ubuntu 18.04, so elementary 5.x

**Comment by esodan** [comment]
Any progress here? 

Because this feature has been integrated into GVls; but is not merged upstream for some reason, I'll push back this MR and remove that integration in order to implement a new way to use your liter as a subprocess

**Review by kgrubb — APPROVED** [review]
I'm also definitely not an expert here, but as far as I can tell, making this installable as a library doesn't really break anything with the current workflow. I'm inclined to merge this and iron out any issues that may arise whenever it starts being used more as a library. Do you have any objections to merging this in its current state @pantor or @btkostner?

**Comment by btkostner** [comment]
From my very basic understanding of vala and desktop stuff, this seems fine

**Pull request merged**